### PR TITLE
[UP-421] Aviation checklist template

### PR DESCRIPTION
**Pull request by 3ace**

### CODE REVIEW

File: templates/aviation-checklist/pdf_aviation_checklist.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+	}
+
+	c.DrawHeader(func(block *creator.Block, args creator.HeaderFunctionArgs) {
+		drawHeader("templates/header.tpl", block, checks.Version, checks.ReleaseDate, args.PageNum, args.TotalPages)
+	})
+	c.DrawFooter(func(block *creator.Block, args creator.FooterFunctionArgs) {
+		drawHeader("templates/footer.tpl", block, checks.Version, checks.ReleaseDate, args.PageNum, args.TotalPages)
+	})
+
+	if err := c.DrawTemplate(mainTpl, checks, tplOpts); err != nil {
```
Comment (adrg) [inline templates/aviation-checklist/pdf_aviation_checklist.go:139]: Please move the drawing of the main template somewhere around line `110` so that it's in the section specified by the comment: `// Draw main content template.`.

File: templates/aviation-checklist/templates/footer.tpl [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <table columns="3">
+        <table-cell border-width-top="2">
+            <paragraph>
+                <text-chunk>Version {{ .Version }} {{ .ReleaseDate }}</text-chunk>
+            </paragraph>
+        </table-cell>
+
+        <table-cell border-width-top="2">
+            <paragraph>
+                <text-chunk>Copyright © 2021 unidoc.io</text-chunk>
```
Comment (adrg) [inline templates/aviation-checklist/templates/footer.tpl:11]: Could add an extra text chunk and make `unidoc.io` a proper external link.

**Review by adrg — COMMENTED** [review]
Looks good. I left a couple of minor observations.

**Comment by 3ace** [comment]
@adrg updated as your suggestion. Thanks

File: templates/aviation-checklist/pdf_aviation_checklist.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+	tplOpts := &creator.TemplateOptions{
+		HelperFuncMap: template.FuncMap{
+			"calcTableHeight": func(check *Checks) float64 {
+				titleSp := c.NewStyledParagraph()
+				chunk := titleSp.SetText(check.Title)
+				chunk.Style.Font = font
+
+				height := titleSp.Height()
+
```
Comment (adrg) [inline templates/aviation-checklist/pdf_aviation_checklist.go:74]: Can you remove this empty line please?

File: templates/aviation-checklist/pdf_aviation_checklist.go [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+func (c *CheckItem) DisplayText() string {
+	if c.Value == nil {
+		return *c.Check
+	}
+
+	// Max line width.
+	maxWidth := 258.0
+
+	stdFont := model.StdFontName(model.HelveticaName)
+	font, err := model.NewStandard14Font(stdFont)
```
Comment (adrg) [inline templates/aviation-checklist/pdf_aviation_checklist.go:212]: Here as well. You can do:
```go
font, err := model.NewStandard14Font(model.HelveticaName)
```

No need for the `stdFont` variable. You are not using it anywhere else.

File: templates/aviation-checklist/pdf_aviation_checklist.go [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+	return *c.Check + "  " + *c.Value
+}
+
+// getRuneWidth calculates rune width based on provided font.
+func getRuneWidth(font *model.PdfFont, r rune) float64 {
+	metrics, bool := font.GetRuneMetrics(r)
+	if !bool {
+		log.Fatal("failed to get width")
+	}
+	return metrics.Wx / 100
```
Comment (adrg) [inline templates/aviation-checklist/pdf_aviation_checklist.go:248]: What does `100` represent? Should it be `1000.0`?

File: templates/aviation-checklist/pdf_aviation_checklist.go [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+		textWidth += getRuneWidth(font, r)
+	}
+
+	if textWidth < maxWidth {
+		availWidth := maxWidth - textWidth
+		separator := '.'
+		sepWidth := getRuneWidth(font, separator)
+		sepCount := int(availWidth / sepWidth)
+
+		// Space to narrow for dot separator, use space
```
Comment (adrg) [inline templates/aviation-checklist/pdf_aviation_checklist.go:228]: Please end sentences with `.`. Also, can you rephrase this comment a bit please? It's not really clear what it means.

**Review by adrg — COMMENTED** [review]
Looks good. I left a couple more observations.

File: templates/aviation-checklist/pdf_aviation_checklist.go (hunk 5 above)
Comment (3ace) [reply templates/aviation-checklist/pdf_aviation_checklist.go]: Yes you are correct, normally this should be 1000.0. It use 100 as a shortcut to include font size in the width calculation. The font size used in the checklist text is 10.

I've rewrite the code to be more clear on this calculation.

**Comment by 3ace** [comment]
@adrg I've updated the PR according to your input

**Review by adrg — APPROVED** [review]
Looks good. Thank you for addressing the review points.
